### ISSUE TITLE
kpcli (Perl) cannot load Crypt::Rijndael

### ISSUE THREAD

**Issue by daemonkeeper** [Homebrew/homebrew-core#84590]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew config
HOMEBREW_VERSION: 3.2.10
ORIGIN: https://github.com/Homebrew/brew
HEAD: ab2118a402a44451b1d7559167b6956dc47cc19a
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 3fd47fb1227134b94e00e9dfe6f38b63ba6f2346
Core tap last commit: 28 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 12.0.5 build 1205
Git: 2.30.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.5.2-x86_64
CLT: 12.5.1.0.1.1623191612
Xcode: 12.5.1


$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Trying to install and run the brew formula for kpcli (https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/kpcli.rb) which pulls in Perl and the Crypt::Rijndael module from CPAN. In the versions selected, they don't seem to fit together due to an ABI mismatch

This was working previously, so I suspect this issue is specific to Big Sur and/or recent dependency changes. 

### What happened (include all command output)?

```
$ kpcli -h
Rijndael.c: loadable library and perl binaries are mismatched (got handshake key 0xc500080, needed 0xc400080)
```

or simply:

```
$ PERL5LIB="/usr/local/Cellar/kpcli/3.6_1/libexec/lib:/usr/local/Cellar/kpcli/3.6_1/libexec/lib/perl5" perl -mCrypt::Rijndael
Rijndael.c: loadable library and perl binaries are mismatched (got handshake key 0xc500080, needed 0xc400080)
```

### What did you expect to happen?

Having a working utility 

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
$ brew reinstall kpcli
```

**Comment by carlocab** [comment]
Can you try
```
brew reinstall -s kpcli
```
?

This will rebuild it from source, which will take a little time.

**Comment by daemonkeeper** [comment]
```
[artoell@us84f437761ce:~] 130 $ brew reinstall -s kpcli

==> Downloading https://cpan.metacpan.org/authors/id/W/WY/WYANT/Mac-Pasteboard-0.011.tar.gz
######################################################################## 100.0%
==> Downloading https://cpan.metacpan.org/authors/id/L/LE/LEONT/Module-Build-0.4231.tar.gz
######################################################################## 100.0%
==> Downloading https://cpan.metacpan.org/authors/id/R/RH/RHANDOM/File-KeePass-2.03.tar.gz
######################################################################## 100.0%
==> Downloading https://cpan.metacpan.org/authors/id/L/LE/LEONT/Crypt-Rijndael-1.15.tar.gz
######################################################################## 100.0%
==> Downloading https://cpan.metacpan.org/authors/id/B/BI/BINGOS/Sort-Naturally-1.03.tar.gz
######################################################################## 100.0%
==> Downloading https://cpan.metacpan.org/authors/id/B/BR/BRONSON/Term-ShellUI-0.92.tar.gz
######################################################################## 100.0%
==> Downloading https://cpan.metacpan.org/authors/id/H/HA/HAYASHI/Term-ReadLine-Gnu-1.37.tar.gz
######################################################################## 100.0%
==> Downloading https://cpan.metacpan.org/authors/id/R/RA/RAZINF/Data-Password-1.12.tar.gz
######################################################################## 100.0%
==> Downloading https://cpan.metacpan.org/authors/id/S/SH/SHLOMIF/Clipboard-0.26.tar.gz
######################################################################## 100.0%
==> Downloading https://cpan.metacpan.org/authors/id/D/DA/DAGOLDEN/Capture-Tiny-0.48.tar.gz
######################################################################## 100.0%
==> Downloading https://downloads.sourceforge.net/project/kpcli/kpcli-3.6.pl
==> Downloading from https://nav.dl.sourceforge.net/project/kpcli/kpcli-3.6.pl
######################################################################## 100.0%
==> Reinstalling kpcli
==> perl Makefile.PL INSTALL_BASE=/usr/local/Cellar/kpcli/3.6_1/libexec
==> make install
==> perl Makefile.PL INSTALL_BASE=/usr/local/Cellar/kpcli/3.6_1/libexec
==> make install
==> perl Makefile.PL INSTALL_BASE=/usr/local/Cellar/kpcli/3.6_1/libexec
==> make install
==> perl Makefile.PL INSTALL_BASE=/usr/local/Cellar/kpcli/3.6_1/libexec
==> make install
==> perl Makefile.PL INSTALL_BASE=/usr/local/Cellar/kpcli/3.6_1/libexec
==> make install
==> perl Makefile.PL INSTALL_BASE=/usr/local/Cellar/kpcli/3.6_1/libexec
==> make install
==> perl Makefile.PL INSTALL_BASE=/usr/local/Cellar/kpcli/3.6_1/libexec
==> make install
==> perl Makefile.PL INSTALL_BASE=/usr/local/Cellar/kpcli/3.6_1/libexec
==> make install
==> perl Build.PL --install_base /usr/local/Cellar/kpcli/3.6_1/libexec
==> ./Build
==> ./Build install
==> perl Makefile.PL INSTALL_BASE=/usr/local/Cellar/kpcli/3.6_1/libexec --includedir=/usr/local/opt/readline/include --libdir=/usr/local/opt/readline/lib
==> make install
🍺  /usr/local/Cellar/kpcli/3.6_1: 144 files, 1.9MB, built in 25 seconds
```

This works, and resolves the issue. Please notice that I did install kpcli and brew only yesterday on a fresh system. I don't _really_ understand brew's package manager logic, but at that point I'd suggest to check if the cached packages (if any) need an update.

**Comment by carlocab** [comment]
`kpcli` was built on an older version of Big Sur -- the issue you're seeing is from Apple upgrading to a newer but incompatible version of Perl.

This should be fixed by #84593.

**Comment by daemonkeeper** [comment]
Awesome, thanks for the quick help!
